File: src/main/java/com/github/shatteredsuite/scrolls/items/NBTVersion.java
package com.github.shatteredsuite.scrolls.items;

public enum NBTVersion {
    VERSION_2(2, null),
    VERSION_1(1, VersionConverter::createScrollInstanceFromV1),
    NONE(0, null);

    public final int nbtSpecifier;
    public final VersionConversion conversion;

    NBTVersion(int nbt_version, VersionConversion conversion) {
        this.nbtSpecifier = nbt_version;
        this.conversion = conversion;
    }
}

File: src/main/kotlin/com/github/shatteredsuite/scrolls/items/NBTVersion.kt
package com.github.shatteredsuite.scrolls.items

import org.bukkit.inventory.ItemStack

enum class NBTVersion(val nbtSpecifier: Int, val conversion: VersionConversion?) {
    VERSION_2(2, null),
    VERSION_1(1, object : VersionConversion {
        override fun convert(stack: ItemStack): ScrollInstance? {
            return VersionConverter.createScrollInstanceFromV1(stack)
        }
    }),
    NONE(0, null);

}